CMakeLists.txt

```diff
@@ -2301,27 +2301,32 @@ endif()
 # FFmpeg 4.x support
 # FFmpeg is multimedia library that can be found http://ffmpeg.org/
 find_package(FFMPEG COMPONENTS libavcodec libavformat libavutil libswresample)
-cmake_dependent_option(FFMPEG "FFmpeg 4.x support" ON "FFMPEG_FOUND;FFMPEG_libavcodec_VERSION VERSION_GREATER_EQUAL 58" OFF)
+# On Windows version numbers are not available!
+if(FFMPEG_libavcodec_VERSION)
+  cmake_dependent_option(FFMPEG "FFmpeg 4.x support" ON "FFMPEG_FOUND;FFMPEG_libavcodec_VERSION VERSION_GREATER_EQUAL 58" OFF)
+else()
+  cmake_dependent_option(FFMPEG "FFmpeg 4.x support" ON "FFMPEG_FOUND" OFF)
+endif()
```

Comment:
I think we should stop using `cmake_dependent_option` (except for options that simply don't exist on that OS).

The reason is that the option will forcibly set to off if the condition is not met, regardless of what the user specified.
That means that if I compile Mixxx with `-DFFMPEG=ON - DMAD=OFF` but forgot to install FFMPEG, Mixxx will be built without it but I don't get any warning (and  wonder why MP3 files won't open).

Instead, this should just be a regular option, with a default value "OFF" for now. We can also consider `"${FFMPEG_FOUND}"` as default but I didn't test that.

The version check can go into the FFMPEG block:

```
if(FFMPEG)
  if(NOT FFMPEG_FOUND)
    message(FATAL_ERROR "FFMPEG was not found") 
  endif()
  if(UNIX AND FFMPEG_libavcodec_VERSION VERSION_LESS 58)
    message(FATAL_ERROR "FFMPEG version needs to be >= 58") 
  endif()
```

We can also consider writing a macro for this to reduce the amount of boilerplate code (not in this PR though).

Reply:
Instead of a custom if block, we can also repeat the `find_package` call and add `REQUIRED` to the arguments.